builder/amazon/common/run_config.go

```diff
@@ -426,6 +426,18 @@ type RunConfig struct {
 	// 10m
 	WindowsPasswordTimeout time.Duration `mapstructure:"windows_password_timeout" required:"false"`
 
+	// Metadata Options
+	// See [Configure IMDS](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html) for details.
+	// A string to enable or disble the IMDS endpoint for an instance. Defaults to enabled.
+	// Accepts either "enabled" or "disabled"
+	HttpEndpoint string `mapstructure:"http_endpoint" required:"false"`
+	// A string to either set the use of IMDSv2 for the instance to optional or required. Defaults to "optional".
+	// Accepts either "optional" or "required"
+	HttpTokens string `mapstructure:"http_tokens" required:"false"`
+	// A numerical value to set an upper limit for the amount of hops allowed when communicating with IMDS endpoints.
+	// Defaults to 1.
+	HttpPutResponseHopLimit int64 `mapstructure:"http_put_response_hop_limit" required:"false"`
+
```

Comment:
You should definitely add some validation in `(c *RunConfig) Prepare(ctx *interpolate.Context) []error` to verify if the options are set with valid values. Like: 

`http_endpoint` should be either "disabled" or "enabled"
`http_put_response_hop_limit` should be between 1 and 64
`http_tokens` should be either "optional" or "required"

Reply:
Setup the validation. Note that with the `http_put_response_hop_limit` one, I couldn't get the check to match 0 specifically, so I did a check of 0<64 and if they don't set the option, the default instantiation of it should be 0, so my sane default checks should catch the 0 in this case and flip it to the default of a 1. Let me know if this is acceptable, or you can see why my != 0 check wasn't working, as I tried it a few different ways and worked out the true/false logic on a piece of paper to make sure I wasn't just being stupid with my assertions.

Reply:
I suggested some changes to the validation where the defaults are set. Could you see if that solves the problem? I think it will.